src/ray/gcs/gcs_server/gcs_node_manager.cc

```diff
@@ -206,14 +209,16 @@ std::shared_ptr<rpc::GcsNodeInfo> GcsNodeManager::RemoveNode(
       std::string type = "node_removed";
       std::ostringstream error_message;
       error_message << "The node with node id: " << node_id
-                    << " and ip: " << removed_node->node_manager_address()
+                    << " and address: " << removed_node->node_manager_address()
+                    << " and node name: " << removed_node->node_name()
                     << " has been marked dead because the detector"
                     << " has missed too many heartbeats from it. This can happen when a "
                        "raylet crashes unexpectedly or has lagging heartbeats.";
       RAY_EVENT(ERROR, EL_RAY_NODE_REMOVED)
               .WithField("node_id", node_id.Hex())
               .WithField("ip", removed_node->node_manager_address())
           << error_message.str();
+      RAY_LOG(ERROR) << error_message.str();
```

Comment:
Why do we need to add this? Does this mean we didn't have log messages before?

Reply:
Yes, there was no error msg

Reply:
Hmm can you verify if it is not printed twice? It is hard to believe there was no log before because I still have seen this message. I assume `RAY_EVENT ` should print the log message?

Reply:
Strangely, I did not observe this in the past, and do not observe this message when testing with this line removed. I also notice this PR only prints once (to both gcs_server.out and gcs_server.err).

The error only appears in driver via this line:
```cpp
      RAY_CHECK_OK(gcs_publisher_->PublishError(node_id.Hex(), *error_data_ptr, nullptr));
```

Not sure if we should remove the above line now that gcs_server stdout + stderr will be printed to driver (as I observed with this line included)?

Reply:
Actually, in this case can you just change the log level to WARNING? Otherwise it will be printed to the driver, and we will print it twice (since we publish the error to the driver, and all ERROR level logs are published to drivers as well)